carbons/carbons_test.go

```diff
@@ -172,3 +172,58 @@ func TestUnwrap(t *testing.T) {
 		})
 	}
 }
+
+var privateValidTestCases = [...]struct {
+	outXML string
+	inXML  string
+}{
+	0: {
```

Comment:
We might also want a test for something like this to make sure it's only the outermost message that is affected:

```
<message xmlns="jabber:client">
<message xmlns="jabber:client"><!-- nothing new inserted here --></message>
<private xmlns="urn:xmpp:carbons:2"></private>
<no-copy xmlns="urn:xmpp:hints"></no-copy>
</message>
```